### PR TITLE
Allow to build only one of the subprojects

### PR DESCRIPTION
**Pull request by stefantakacs**
I personally prefer not to be forced to build motor.

### CODE REVIEW

**Comment by filiatra** [comment]
let's keep this in mind, and merge when cronjob nightly scripts are updated..
Also, motor most likely depends on unsupported..

**Comment by stefantakacs** [comment]
Yes, we are not in a hurry.

**Comment by stefantakacs** [comment]
@codewerfer Please add GISMO_MOTOR=ON where GISMO_UNSUPPORTED=ON within server scripts.

**Comment by mmoelle1** [comment]
Let me briefly react on this. The MOTOR project is over and the number of MOTOR partners that continue developing in G+Smo is moderate. It’s basically UWB Pilzen, JKU, TU Kaiserslautern and TU Delft. The two latter ones already know about the new layout of G+Smo extensions. JKU, that’s you and UWB just needs one mail to three people (Bohumir, Hana, and Eva) and they can update their checkouts.

**Comment by codewerfer** [comment]
OK, so revert my last commit.

**Comment by stefantakacs** [comment]
@codewerfer  Did you update the scripts?

**Comment by stefantakacs** [comment]
Btw, is there a possibility to send out an email via github?

**Comment by stefantakacs** [comment]
I now suggest some rearrangement which make the file clearer.

I hope this still works!?

**Comment by codewerfer** [comment]
sorry, had to do some support.
No, not finished with all servers yet - only my machine.

**Comment by stefantakacs** [comment]
Please write me here, when you are done

**Comment by codewerfer** [comment]
Sorry, https://github.com/gismo/gsElasticity/issues/10 costs my some time.
It is similar to this one - and that what we talked about how to prevent to clone **motor** submodule.

Btw. you can squash and merge this branch. It will not build motor on machines where it is not set yet, but the build will fail on most of this machines because of https://github.com/gismo/gsElasticity/issues/10 anyway.

File: CMakeLists.txt [hunk 1]
```diff
@@ -306,14 +306,6 @@ if(GISMO_BUILD_LIB)
...
 endif()
 
-if(GISMO_ELASTICITY)
-  gismo_fetch_module(gsElasticity)
-endif()
-
-if(GISMO_EXASTENCILS)
-  gismo_fetch_module(gsExaStencils)
-endif()
-
```
Comment (filiatra) [inline CMakeLists.txt]: I think there was a reason why this was in this position, does everything still work after this commit ?

Comment (stefantakacs) [reply CMakeLists.txt]: I have tried to invoke
```
cmake . -DGISMO_ELASTICITY=On
cmake . -DGISMO_EXASTENCILS=On
make
```
in my build folder and it worked (up to the bugs the code had/has anyway).

File: CMakeLists.txt [hunk 2]
```diff
@@ -381,10 +373,25 @@ if (NOT gismo_optionals STREQUAL "")
...
 endif()
 
+if(GISMO_ELASTICITY)
+  gismo_fetch_module(gsElasticity)
+endif()
+
+if(GISMO_EXASTENCILS)
+  gismo_fetch_module(gsExaStencils)
+endif()
+
```
Comment (codewerfer) [inline CMakeLists.txt]: @filiatra this code you mentioned is still there

Comment (stefantakacs) [reply CMakeLists.txt]: its about the order. that might matter.

**Comment by stefantakacs** [comment]
Can it be true that (after downloading the submodule gsElasticity) that submodule is included even if it is deactivated in cmake?

Maybe because its name starts with gs and there is a corresponding rule?

**Comment by codewerfer** [comment]
> Can it be true that (after downloading the submodule gsElasticity) that submodule is included even if it is deactivated in cmake?

Yes. That's the problem we get on @rschneckenleitner's computer.
Workaround to solve this:
`git submodule deinit extensions/gsElasticity`
 
> Maybe because its name starts with gs and there is a corresponding rule?

I do not know. If so, it will be also with `gsExaStencils`.
https://github.com/gismo/gismo/blob/5e83df1d3180291700b66e80a2209a65f593b6b0/cmake/gismoUse.cmake#L78-L88
A better solution for this would be nice.

----
If changed the wiki therefore, that somebody should `git clone <gismo>` without `--recusrive-submodules` and activate - and clone - them via the `-DGISMO_<submodule>` option.

**Comment by codewerfer** [comment]
> Please write me here, when you are done

@stefantakacs Have updated all cdash servers I have access to.
Have `submodule deinit` all submodules. Have added `-DGISMO_MOTOR=ON` to build script.

**Comment by stefantakacs** [comment]
Then let's try this.

**Pull request merged**